src/commands/cmd_txn.cc

```diff
@@ -79,13 +91,37 @@ class CommandExec : public Commander {
       conn->ExecuteCommands(conn->GetMultiExecCommands());
       s = storage->CommitTxn();
     }
-    conn->ResetMultiExec();
     return s;
   }
 };
 
+class CommandWatch : public Commander {
+ public:
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    if (conn->IsFlagEnabled(Connection::kMultiExec)) {
+      *output = Redis::Error("ERR WATCH inside MULTI is not allowed");
+      return Status::OK();
+    }
+
+    svr->WatchKey(conn, std::vector<std::string>(args_.begin() + 1, args_.end()));
+    *output = Redis::SimpleString("OK");
+    return Status::OK();
+  }
+};
+
+class CommandUnwatch : public Commander {
+ public:
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
```

Comment:
Should unwatch need `kMultiExec` support?

Reply:
It seems weird to me that `unwatch` does not fail in multi in Redis, e.g. in redis-cli
```
MULTI
WATCH # error
...
```
```
MULTI
UNWATCH # ok
...
```

Reply:
It's funny. Seems ok